File: ptolemy/python/ptolemy/models/enums.py
"""Enums."""

from enum import StrEnum


class ApiKeyPermissionEnum(StrEnum):
    """API Key Permissions Enum."""

    READ_ONLY = "READ_ONLY"
    WRITE_ONLY = "WRITE_ONLY"
    READ_WRITE = "READ_WRITE"


class UserStatusEnum(StrEnum):
    """User Status Enum."""

    ACTIVE = "ACTIVE"
    SUSPENDED = "SUSPENDED"

class WorkspaceRoleEnum(StrEnum):
    """Workspace Role enum."""

    USER = "USER"
    MANAGER = "MANAGER"
    ADMIN = "ADMIN"

File: ptolemy/src/api/models/auth/enums.rs
use crate::{
    api::generated::auth_schema::sql_types::{ApiKeyPermission, UserStatus, WorkspaceRole},
    define_enum,
};
use diesel::deserialize::FromSql;
use diesel::serialize::{IsNull, Output, ToSql};
use diesel::{
    AsExpression, FromSqlRow,
    {pg::Pg, pg::PgValue},
};
use juniper::GraphQLEnum;
use std::io::Write;

define_enum!(WorkspaceRoleEnum, WorkspaceRole, [User, Manager, Admin]);
define_enum!(UserStatusEnum, UserStatus, [Active, Suspended]);
define_enum!(
    ApiKeyPermissionEnum,
    ApiKeyPermission,
    [ReadOnly, WriteOnly, ReadWrite]
);
